manager/controllers/app/plotter_generator.go

```diff
@@ -201,6 +204,55 @@ func (p *PlotterGenerator) addStep(element *ResolvedEdge, datasetID string, api
 	return steps
 }
 
+// Create a new asset and register it to the catalog. Used when the IsNewDataSet flag is true
+func (p *PlotterGenerator) HandleNewAsset(item *DataInfo, application *app.FybrikApplication, selection *Solution,
+	plotterSpec *app.PlotterSpec) error {
+	var err error
+	if item.DataDetails.Details.DataFormat != "" {
+		return nil
+	}
+	var sinkDataStore *app.DataStore
+	selectionLen := len(selection.DataPath)
+	element := selection.DataPath[selectionLen-1]
+	// allocate storge
+	if sinkDataStore, err = p.AllocateStorage(item, element.Sink.Connection, &element.StorageAccount); err != nil {
+		p.Log.Error().Err(err).Str(logging.DATASETID, item.Context.DataSetID).Msg("Storage allocation failed")
+		return err
+	}
+
+	item.DataDetails.Details.DataFormat = sinkDataStore.Format
```

Comment:
Not only data format should be populated

Reply:
@shlomitk1 we fixed that. Please take a look. Thanks